### PR TITLE
Update reference_linter.yaml

### PR DESCRIPTION
**Pull request by pdehaan**
Looks like we moved pendingTranslations.ftl into locales-pending* in #3187

# References: 
Jira: MNTOR-
Figma: 

# Description

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Comment by pdehaan** [comment]
🤷 Not 💯% sure this is correct, but trying to wrap my head around .github/workflows/reference_linter.yaml and if those were stale references. But I think we also have that `ci/circleci: lint-l10n` check which is validating pending .ftl changes via <kbd>moz-fluent-lint ./locales-pending/ --config .github/linter_config.yml</kbd> anyways.

File: .github/workflows/reference_linter.yaml [hunk 1]
```diff
@@ -3,14 +3,14 @@ on:
   push:
     paths:
       - 'locales/en/**.ftl'
-      - 'pendingTranslations.ftl'
+      - 'locales-pending/*.ftl'
     branches:
       - main
       - localization
```
Comment (flodolo) [inline .github/workflows/reference_linter.yaml:9]: We should remove this, it's not used anymore

File: .github/workflows/reference_linter.yaml [hunk 2]
```diff
@@ -3,14 +3,14 @@ on:
...
   pull_request:
     paths:
       - 'locales/en/**.ftl'
```
Comment (flodolo) [inline .github/workflows/reference_linter.yaml:12]: We should add this workflow to the paths triggering the workflow.

**Review by flodolo — CHANGES_REQUESTED** [review]

File: .github/workflows/reference_linter.yaml (hunk 1 above)
Comment (Vinnl) [reply .github/workflows/reference_linter.yaml:9]:
```suggestion
```

**Comment by Vinnl** [comment]
Oh good call, with flod's suggestions this looks good.

> But I think we also have that ci/circleci: lint-l10n check which is validating pending .ftl changes via moz-fluent-lint ./locales-pending/ --config .github/linter_config.yml anyways.

Hmm, I wasn't aware of this job :sweat_smile: We should probably remove that (or the GitHub workflow), and I suppose add `compare-locales` to the GitHub workflow?

**Comment by flodolo** [comment]
> Hmm, I wasn't aware of this job 😅 We should probably remove that (or the GitHub workflow), and I suppose add `compare-locales` to the GitHub workflow?

I'd keep this workflow, since this is the way we run linters across different products.

No real need to run compare-locales, we have plenty of linters around besides this one.

**Comment by Vinnl** [comment]
OK, then I'd do it the other way around - update that job to include the pending strings, and remove the GH workflow :)

**Comment by flodolo** [comment]
> OK, then I'd do it the other way around - update that job to include the pending strings, and remove the GH workflow :)

Wait, I asked to keep the GitHub workflow, not to remove it 🤔

**Comment by Vinnl** [comment]
Ah! My initial suggestion was to remove the redundant CircleCI job - the linter currently runs twice, once in CircleCI and once in GitHub Actions. Once is probably enough :sweat_smile:

File: .github/workflows/reference_linter.yaml [hunk 3]
```diff
@@ -3,14 +3,14 @@ on:
...
-      - 'pendingTranslations.ftl'
+      - 'locales-pending/*.ftl'
     branches:
       - main
       - localization
```
Comment (pdehaan) [inline .github/workflows/reference_linter.yaml:16]: @flodolo @Vinnl Remove this too?
```suggestion
```

Comment (flodolo) [reply .github/workflows/reference_linter.yaml:15]: Yep

File: .github/workflows/reference_linter.yaml [hunk 4]
```diff
@@ -3,17 +3,15 @@ on:
   push:
     paths:
       - 'locales/en/**.ftl'
```
Comment (flodolo) [inline .github/workflows/reference_linter.yaml:5]:
```suggestion
      - '.github/workflows/reference_linter.yaml'
      - 'locales/en/**.ftl'
```

File: .github/workflows/reference_linter.yaml [hunk 5]
```diff
@@ -3,17 +3,15 @@ on:
...
-      - 'pendingTranslations.ftl'
+      - 'locales-pending/*.ftl'
     branches:
       - main
-      - localization
   pull_request:
     paths:
       - 'locales/en/**.ftl'
```
Comment (flodolo) [inline .github/workflows/reference_linter.yaml:11]:
```suggestion
      - '.github/workflows/reference_linter.yaml'
      - 'locales/en/**.ftl'
```

**Review by flodolo — APPROVED** [review]

**Review by Vinnl — APPROVED** [review]
LGTM! Though good idea to also run this if the job itself gets modified :)

**Pull request merged**